libs/gorules/rule_empty_slices.go

```diff
@@ -0,0 +1,15 @@
+package gorules
+
+import "github.com/quasilyte/go-ruleguard/dsl"
+
+// DeclaringEmptySlices detects empty slice declarations and suggests using nil slices.
+// See: https://go.dev/wiki/CodeReviewComments#declaring-empty-slices
+func DeclaringEmptySlices(m dsl.Matcher) {
+	m.Match(`$x := []$t{}`).
```

Comment:
nit: This does not account for slice of slices: `var queue [][]int = [][]int{{}}` (in struct_info.go) or likely slice of maps (eg: `[]map[string]any{{}}`).

Should we also account for them?

Reply:
I would say it is out of scope of this change